src/gentropy/dataset/study_locus.py

```diff
@@ -453,6 +457,60 @@ def clump(self: StudyLocus) -> StudyLocus:
         )
         return self
 
+    def get_gnomad_matrix(
+        self: StudyLocus, session: Session, study_index: StudyIndex, window_size: int
+    ) -> tuple[DataFrame, np.ndarray]:
+        """Extract hail matrix index and block matrix from StudyLocus rows.
+
+        Args:
+            session (Session): Spark session
+            study_index (StudyIndex): Study index
+            window_size (int): Window size to extract from gnomad matrix
+
+        Returns:
+            tuple[DataFrame, np.ndarray]: Returns the index of the gnomad matrix and the gnomad block matrix
+        """
+        _df = (
+            self.df.join(study_index.get_major_pop(), on="studyId", how="left")
+            .withColumn("start", f.col("position") - (window_size / 2))
+            .withColumn("end", f.col("position") + (window_size / 2))
+            .alias("_df")
+        )
+        _pop = _df.collect()[0]["ldPopulationStructure"]
```

Comment:
When I originally wrote the code to extract the Gnomad LD indices & block matrices for benchmarking, I kept running into issues trying to parallelise the step of extracting the block matrices for each locus. So I thought it was best to get it working for a single study-locus first. I might have have just been doing something wrong though.

Reply:
Not sure that this is relevant but we disused a lot the these issues (within OTG and with @d0choa ) and concluded the following:
1) All functions related to FM (LD block matrix extraction, Z-score extraction, FM itself) will be applicable to one study-locus at a time. Of course, there is a room for optimisation, but we need to get it working on one study-locus first. The parallelisation is possible only on the level of study-loci (one study locus per cpu).
2) For each study-locus we will assume only ONE major population to extract LD (the most prevalent population), or, in case of absence the information, nfe.